wrapper/src/main/java/software/amazon/jdbc/plugin/iam/IamAuthConnectionPlugin.java

```diff
@@ -87,13 +89,45 @@ public class IamAuthConnectionPlugin extends AbstractConnectionPlugin {
   private final TelemetryGauge cacheSizeGauge;
   private final TelemetryCounter fetchTokenCounter;
 
+  private IamTokenUtility iamTokenUtility = new RegularRdsUtility();
+
   public IamAuthConnectionPlugin(final @NonNull PluginService pluginService) {
+
+    this.checkRequiredDependencies();
+
     this.pluginService = pluginService;
     this.telemetryFactory = pluginService.getTelemetryFactory();
     this.cacheSizeGauge = telemetryFactory.createGauge("iam.tokenCache.size", () -> (long) tokenCache.size());
     this.fetchTokenCounter = telemetryFactory.createCounter("iam.fetchToken.count");
   }
 
+  private void checkRequiredDependencies() {
+    try {
+      // RegularRdsUtility requires AWS Java SDK RDS v2.x to be presented in classpath.
+      Class.forName("software.amazon.awssdk.services.rds.RdsUtilities");
+    } catch (final ClassNotFoundException e) {
+      ClassNotFoundException lastException = e;
+      // If SDK RDS isn't presented, try to check required dependency for LightRdsUtility.
+      try {
+        // LightRdsUtility requires "software.amazon.awssdk.auth"
+        // and "software.amazon.awssdk.http-client-spi" libraries.
+        Class.forName("software.amazon.awssdk.http.SdkHttpFullRequest");
+        Class.forName("software.amazon.awssdk.auth.signer.params.Aws4PresignerParams");
+
+        // Required libraries are presented. Use lighter version of RDS utility.
+        iamTokenUtility = new LightRdsUtility();
+        lastException = null;
+
+      } catch (final ClassNotFoundException ex) {
+        throw new RuntimeException(Messages.get("IamAuthConnectionPlugin.javaSdkNotInClasspath"), ex);
+      }
+
+      if (lastException != null) {
```

Comment:
Is this check necessary? If this is reached, wouldn't lastException always be null, since it's set on line 119?

Reply:
Agree, it's not necessary.